guides/session.rst

```diff
@@ -1,76 +1,116 @@
+tabs).
+
+The first thing to do with your session is to start it. Nothing can be done
+with it before starting it.
 
 .. code-block:: php
 
-    // init session:
+    // Choose a Mink driver. More about it in later chapters.
+    $driver = new \Behat\Mink\Driver\GoutteDriver();
+
     $session = new \Behat\Mink\Session($driver);
 
-    // start session:
+    // start the session
     $session->start();
 
 .. note::
 
-  As you can see, the first argument to the session (``$driver``) is just
-  a simple driver instance, which we created in the previous chapter.
+    The first argument to the session constructor is a driver object. Drivers
+    are the way the Mink abstraction layer works. You will discover more
+    about the available drivers in a :doc:`later chapter </guides/drivers>`.
+
+.. caution::
 
-``start()`` call is required in order to configure the browser emulator or
-controller to be fully functional.
+    Although Mink does its best on removing browser differences between different
+    browser emulators, it can't do much in some cases. See the :ref:`driver-feature-support`
+    to see which features are supported by each driver.
 
 Basic Browser Interaction
-~~~~~~~~~~~~~~~~~~~~~~~~~
+-------------------------
 
-After you've instantiated the ``$session`` object, you can control the actual
-browser emulator with it:
+The first thing to do when using your session is to open a page with it.
+Just after starting, the session is not on any page (in a real browser, you
+would on the ``about:blank`` page), and calling any other action is likely
+to fail.
 
 .. code-block:: php
 
-    // open some page in browser:
     $session->visit('http://my_project.dev/some_page.php');
 
-    // get the current page URL:
-    echo $session->getCurrentUrl();
+.. note::
 
-    // get the response status code:
-    echo $session->getStatusCode();
+    Mink is primarily designed to be used for testing websites. To allow
+    covering error pages as well, the ``Session::visit`` method does not
+    consider error status codes as invalid. It will not throw an exception
+    in this case. You will need to check whether the response was a success
+    or an error. It will only throw an exception when Mink cannot load the
+    page (network error, ...). For cases, when driver does not support status
+    code retrieval, it might be possible to assert text on the page itself
+    to determine whether it is an error page or no.
 
-    // get page content:
-    echo $session->getPage()->getContent();
+Interacting with the Page
+-------------------------
 
-    // open another page:
-    $session->visit('http://my_project.dev/second_page.php');
+The session gives you access to the page through the ``Session::getPage``
+method. This allows you to :doc:`traverse </guides/traversing-pages>` and
+:doc:`manipulate it </guides/manipulating-pages>`. The next chapters are
+covering the page API in depth.
```

Comment:
The next chapters cover the page API in depth. Most of what you'll do with Mink will use this object, but you can continue reading to learn more about the Session.